tsdb/engine/tsm1/engine.go

```diff
@@ -1161,6 +1161,12 @@ func (e *Engine) createVarRefIterator(opt influxql.IteratorOptions, aggregate bo
 				return err
 			}
 
+			if opt.Limit > 0 || opt.Offset > 0 || opt.SLimit > 0 || opt.SOffset > 0 {
```

Comment:
The tag sets are ordered by key still.  This just order the series keys within each tagset.  

Do you know of a query that could have this non-determinism?